chapters/part3/fairness.md

```diff
@@ -0,0 +1,731 @@
+
+1. The first step is to obtain the pretained model for which you would like to reduce binary gender bias. In this case, we will use [AllenNLP's public large RoBERTA model for SNLI](https://storage.googleapis.com/allennlp-public-models/snli-roberta.2021-03-11.tar.gz).
+
+2. `BiasMitigatorApplicator` works by applying a bias mitigator of your choice directly after the static embedding layer of the pretrained RoBERTA model during finetuning and evaluation. This reduces the model's ability to propagate gender biases across words in input sentences through contextualization.
+
+Below is an [example config file](https://github.com/allenai/allennlp-models/blob/main/training_config/pair_classification/binary_gender_bias_mitigated_snli_roberta.jsonnet) for finetuning a pretrained RoBERTA model for SNLI using `BiasMitigatorApplicator` with linear bias mitigation and a two-means bias direction. As mentioned previously, `BiasMitigatorApplicator` simply wraps the pretrained model. `two_means` requires a `seed_word_pairs_file` and tokenizer to tokenize the words in said file. `seed_word_pairs_file` must follow the format in the example: `[["woman", "man"], ["girl", "boy"], ["she", "he"], ...]`. And that's it!
+
+```python
+local transformer_model = "roberta-large";
+local transformer_dim = 1024;
+
+{
+  "dataset_reader":{
+    "type": "snli",
+    "tokenizer": {
+      "type": "pretrained_transformer",
+      "model_name": transformer_model,
+      "add_special_tokens": false
+    },
+    "token_indexers": {
+      "tokens": {
+        "type": "pretrained_transformer",
+        "model_name": transformer_model,
+        "max_length": 512
+      }
+    }
+  },
+  "train_data_path": "https://allennlp.s3.amazonaws.com/datasets/snli/snli_1.0_train.jsonl",
+  "validation_data_path": "https://allennlp.s3.amazonaws.com/datasets/snli/snli_1.0_dev.jsonl",
+  "model": {
+    "type": "allennlp.fairness.bias_mitigator_applicator.BiasMitigatorApplicator", 
+    "base_model": {
+      "_pretrained": {
+        "archive_file": "https://storage.googleapis.com/allennlp-public-models/snli-roberta.2021-03-11.tar.gz",
+        "module_path": "",
+        "freeze": false
+      }
+    },
+    "bias_mitigator": {
+      "type": "linear",
+      "bias_direction": {
+        "type": "two_means",
+        "seed_word_pairs_file": "https://raw.githubusercontent.com/tolga-b/debiaswe/4c3fa843ffff45115c43fe112d4283c91d225c09/data/definitional_pairs.json",
```

Comment:
Are there a lot of magical seed files? Can we "ship" them in some way, for example, but allowing the `seed_word_pairs_file` to have magical strings that map to download locations like the one you have here? That would help a lot with discoverability.

Reply:
It would help with discoverability, but I'm also worried about transparency. I'm unsure if users will take the time to find the download location to which magical strings map.

Reply:
I would expect most users to not care about the download location at all, regardless of whether we make it easy or hard to find.

Reply:
Sure, but I want users to ideally follow the link and look at what seed words are being used. I also don't like the idea of implicitly endorsing any set of seed words through magical strings; in this case, it's clear that we're just using someone else's words while not necessarily endorsing it.